### PR TITLE
Audio

### PR DESCRIPTION
**Pull request by MiguelZh**

### CODE REVIEW

File: Drakhtar/SDLAudioManager.cpp [hunk 1]
```diff
@@ -134,4 +145,14 @@ void SDLAudioManager::pauseMusic() {
...
-}
\ No newline at end of file
+}
+
+void SDLAudioManager::FadeOutChannel(int channel, int ticks)
+{
+	Mix_FadeOutChannel(channel, ticks);
+}
+
+void SDLAudioManager::FadeOutMusic( int ticks)
```
Comment (kyranet) [inline Drakhtar/SDLAudioManager.cpp]:
```suggestion
void SDLAudioManager::FadeOutMusic(int ticks)
```

File: Drakhtar/AudioManager.h [hunk 2]
```diff
@@ -13,6 +13,7 @@ class AudioManager
 	// sound effects
 	virtual bool loadSound(int tag, string fileName) = 0;
 	virtual int playChannel(int tag, int loops, int channel = -1) = 0;
+	virtual int playChannelTimed(int tag, int loops, int channel , int ticks) = 0;
```
Comment (kyranet) [inline Drakhtar/AudioManager.h]:
```suggestion
	virtual int playChannelTimed(int tag, int loops, int channel, int ticks) = 0;
```

File: Drakhtar/SDLAudioManager.cpp [hunk 3]
```diff
@@ -75,6 +75,17 @@ int SDLAudioManager::playChannel(int tag, int loops, int channel) {
 	}
 }
 
+int SDLAudioManager::playChannelTimed(int tag, int loops, int channel,int ticks)
+{
+	Mix_Chunk* chunk = chunks_[tag];
```
Comment (kyranet) [inline Drakhtar/SDLAudioManager.cpp]:
```cpp
Mix_Chunk* chunk = chunks_[tag];
return chunk != nullptr ? Mix_PlayChannelTimed(channel, chunk, loops, ticks) : -1;
```

File: Drakhtar/SDLAudioManager.cpp [hunk 4]
```diff
@@ -120,7 +131,7 @@ void SDLAudioManager::playMusic(int tag, int loops) {
 	}
 }
 
-int SDLAudioManager::setMusicVolume(int volume) {
+int SDLAudioManager::setMusicVolume(int volume) { // volume = 2 is quite low already, play with that number
```
Comment (kyranet) [inline Drakhtar/SDLAudioManager.cpp]: What does "play with that number" exactly mean?

File: Drakhtar/SDLAudioManager.h [hunk 5]
```diff
@@ -5,6 +5,9 @@
 #include "checkML.h"
 class SDLAudioManager : public AudioManager
 {
+//IMPORTANT NOTES:
+	//Channel means one sound, meaning for example a footstep
```
Comment (kyranet) [inline Drakhtar/SDLAudioManager.h]:
```suggestion
// Channel means one sound, meaning for example a footstep
```

Comment (kyranet) [reply Drakhtar/SDLAudioManager.h]: Same for the line below

File: Drakhtar/SDLAudioManager.h [hunk 6]
```diff
@@ -16,12 +19,14 @@ class SDLAudioManager : public AudioManager
 	// sound effects
 	virtual bool loadSound(int tag, string fileName);
 	virtual int playChannel(int tag, int loops, int channel);
+	virtual int playChannelTimed(int tag, int loops, int channel,int ticks);
```
Comment (kyranet) [inline Drakhtar/SDLAudioManager.h]:
```suggestion
	virtual int playChannelTimed(int tag, int loops, int channel, int ticks);
```

File: Drakhtar/SDLAudioManager.h [hunk 7]
```diff
@@ -16,12 +19,14 @@ class SDLAudioManager : public AudioManager
...
 	virtual void pauseChannel(int channel);
 	virtual void resumeChanne(int channel);
 	virtual void haltChannel(int channel);
 	virtual int setChannelVolume(int volume, int channel);
 	virtual int channels();
 
+
```
Comment (kyranet) [inline Drakhtar/SDLAudioManager.h]:
```suggestion
```

File: Drakhtar/State.h [hunk 8]
```diff
@@ -26,10 +27,16 @@ class State : public GameState
...
-	
 	void boton() { cout << "boton"; }
+	SDLAudioManager audioManager;
 public:
 	State(Game* game, SDL_Renderer* renderer);
 	virtual ~State();
+	virtual void addGameObject(GameObject* gameObject);
+	virtual void removeGameObject(GameObject* gameObject);
+
+	//temporal
```
Comment (kyranet) [inline Drakhtar/State.h]:
```suggestion
	// temporary
```

File: Drakhtar/State.h [hunk 9]
```diff
@@ -26,10 +27,16 @@ class State : public GameState
...
+	void playASound(int tag, int loop, int channels);
```
Comment (kyranet) [inline Drakhtar/State.h]: This should be named "playSound"

File: Drakhtar/SDLAudioManager.cpp (hunk 4 above)
Comment (MiguelZh) [reply Drakhtar/SDLAudioManager.cpp]: It means that if volume is 2, the sound is low, so based on that, anything higher than the value "2" will be higher volume.
In short, if volume is 2, you will barely hear the music

File: Drakhtar/State.h (hunk 9 above)
Comment (MiguelZh) [reply Drakhtar/State.h]: will do

**Pull request merged**